lib/jaeger/client/udp_sender.rb

```diff
@@ -12,13 +12,14 @@ def initialize(host:, port:, encoder:, logger:)
         @logger = logger
 
         transport = Transport.new(host, port)
-        protocol = ::Thrift::CompactProtocol.new(transport)
+        @protocol_class = ::Thrift::CompactProtocol
+        protocol = @protocol_class.new(transport)
         @client = Jaeger::Thrift::Agent::Client.new(protocol)
       end
 
       def send_spans(spans)
-        batch = @encoder.encode(spans)
-        @client.emitBatch(batch)
+        batches = @encoder.encode_limited_size(spans, @protocol_class, 8_000)
```

Comment:
Why 8_000? Also extract to a constant

Reply:
Agreed, it should be a constant.
As for why it's 8000, it's because the default UDP message size for Macs is 9216 and I want to stay enough below it to account for any overhead. (Most OS's seem to have the UDP message size set to 65536. I don't know why Apple has it so low for Macs.)

I did try to find a way to query the OS for the max UDP size in order to avoid having a magic number, but I couldn't find a good platform-independent solution in Ruby. If you know one, please let me know.

Reply:
I think it should be configurable and default should be 65K like in the jaeger java/other tracers: https://github.com/jaegertracing/jaeger-client-java/blob/6548879da36eb26d712037866b90685d1209ff1f/jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/reporters/protocols/ThriftUdpTransport.java#L34 I want to follow the other jaeger tracers as closely as possible.